src/components/button-icon/button-icon.component.js

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Events from '../../utils/helpers/events';
+import Icon from '../icon/icon.component';
+import StyledButtonIcon from './button-icon.style';
+
+const ButtonIcon = ({ onAction, type, ...rest }) => {
+  const onKeyDown = (e) => {
+    if (Events.isEnterKey(e) || Events.isSpaceKey(e) || Events.isEscKey(e)) {
+      e.preventDefault();
+      onAction(e);
+    } else {
+      e.stopPropagation();
+    }
+  };
+
+  const buttonRef = React.createRef();
+
+  const handleOnAction = (e) => {
+    e.preventDefault();
+    onAction(e);
+  };
+
+  return (
+    <StyledButtonIcon
+      { ...rest }
+      onKeyDown={ onKeyDown }
+      onClick={ handleOnAction }
+    >
+      <Icon
+        type={ type }
+        ref={ buttonRef }
```

Comment:
could you tell me why do we need this ref?

Reply:
it is redundant, I left it there by accident, apologies.
fixed in: b996854